File: app/src/main/java/com/daviddf/geeklabtest/Not.java
package com.daviddf.geeklabtest;

import android.app.Application;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.os.Build;

public class Not extends Application {
    public static final String CHANNEL_1_ID = "GeekLab";

    @Override
    public void onCreate() {
        super.onCreate();
        createNotificationChannels();
    }
    private void createNotificationChannels() {
        NotificationChannel channel1 = new NotificationChannel(
                CHANNEL_1_ID,
                "GeekLab",
                NotificationManager.IMPORTANCE_HIGH
        );
        channel1.setDescription("Notifiaciones GeekLab");

        NotificationManager manager = getSystemService(NotificationManager.class);
        manager.createNotificationChannel(channel1);

    }
}

File: app/src/main/java/com/daviddf/geeklab/notification/Not.kt
package com.daviddf.geeklab.notification

import android.app.Application
import android.app.NotificationChannel
import android.app.NotificationManager
import android.os.Build
import androidx.annotation.RequiresApi

class Not : Application() {

    @RequiresApi(Build.VERSION_CODES.O)
    override fun onCreate() {
        super.onCreate()
        createNotificationChannels()
    }

    @RequiresApi(Build.VERSION_CODES.O)
    private fun createNotificationChannels() {
        val channel1 = NotificationChannel(
            CHANNEL_1_ID,
            "GeekLab",
            NotificationManager.IMPORTANCE_HIGH
        ).apply {
            description = "Notificaciones GeekLab"
        }

        val manager = getSystemService(NotificationManager::class.java)
        manager.createNotificationChannel(channel1)
    }

    companion object {
        const val CHANNEL_1_ID = "GeekLab"
    }
}
